pkg/ai/amazonsagemaker.go

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright 2023 The K8sGPT Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ai
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"encoding/json"
+	"log"
+
+	"github.com/fatih/color"
+	"github.com/k8sgpt-ai/k8sgpt/pkg/cache"
+	"github.com/k8sgpt-ai/k8sgpt/pkg/util"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sagemakerruntime"
+)
+
+type SageMakerAIClient struct {
+	client      *sagemakerruntime.SageMakerRuntime
+	language    string
+	model       string
+	temperature float32
+	endpoint    string
+}
+
+const (
+	// SageMaker completion parameters
+	maxNewTokens = 256
+	top_P        = 0.9
```

Comment:
This would really cause many hallucinations - I suggest for credibility we lower this. Perhaps even parameterise it?

Reply:
An example of using the passed in temperature https://github.com/k8sgpt-ai/k8sgpt/blob/main/pkg/ai/azureopenai.go#L46